m4/pbs_default_scheduler.m4

```diff
@@ -0,0 +1,51 @@
+
+#
+#  Copyright (C) 1994-2017 Altair Engineering, Inc.
+#  For more information, contact Altair at www.altair.com.
+#   
+#  This file is part of the PBS Professional ("PBS Pro") software.
+#  
+#  Open Source License Information:
+#   
+#  PBS Pro is free software. You can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License as published by the Free 
+#  Software Foundation, either version 3 of the License, or (at your option) any 
+#  later version.
+#   
+#  PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY 
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+#  PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+#   
+#  You should have received a copy of the GNU Affero General Public License along 
+#  with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   
+#  Commercial License Information: 
+#  
+#  The PBS Pro software is licensed under the terms of the GNU Affero General 
+#  Public License agreement ("AGPL"), except where a separate commercial license 
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#   
+#  Altair’s dual-license business model allows companies, individuals, and 
+#  organizations to create proprietary derivative works of PBS Pro and distribute 
+#  them - whether embedded or bundled with other software - under a commercial 
+#  license agreement.
+#  
+#  Use of Altair’s trademarks, including but not limited to "PBS™", 
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's 
+#  trademark licensing policies.
+#
+
+AC_DEFUN([PBS_AC_DEFAULT_SCHEDULER],
+[
+  AC_MSG_CHECKING([for Default scheduler name])
+  AC_ARG_VAR([PBS_DFLT_SCHED_NAME], [Specifies the PBS default scheduler name.])
+  AS_IF([test "x$PBS_DFLT_SCHED_NAME" = "x"],
```

Comment:
Well, The m4 macro function was added with intention to move all default macros(like name, cycle legnth etc) of scheduler to one place and same can be accessed by adding pbs_config.h. 

The m4 macro function is attempt clean-up the code.  
 
And, regarding configuring default sched name; yes its going to bad idea if we allow to configure.  We end up in upgrades issues. 

In latest commit we disabled configure option.